fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java

```diff
@@ -1093,6 +1119,14 @@ public void replayDropBackend(Backend backend) {
         final Cluster cluster = GlobalStateMgr.getCurrentState().getCluster();
         if (null != cluster) {
             cluster.removeBackend(backend.getId());
+
+            // drop it from warehouse
+            Warehouse warehouse = GlobalStateMgr.getCurrentWarehouseMgr().
+                    getWarehouse(WarehouseManager.DEFAULT_WAREHOUSE_NAME);
+            if (warehouse != null) {
+                warehouse.getAnyAvailableCluster().dropNode(backend.getId());
```

Comment:
should be `replayDropNode`? this will be only focus on warehouse restore its internal state, not triggering starmgr to drop the node again because starmgr itself have edit log and can replay to its final state.